src/runtime/storage.ts

```diff
@@ -26,23 +29,42 @@ const TWO_WEEKS_IN_MILLIS = 2 * 604800000;
  */
 export class Storage {
   private readonly win_: Window;
+  private readonly pageConfig_: PageConfig;
   private readonly values_: {[key: string]: Promise<string | null>};
 
-  constructor(win: Window) {
+  constructor(win: Window, pageConfig: PageConfig) {
     this.win_ = win;
-
+    this.pageConfig_ = pageConfig;
     this.values_ = {};
   }
 
-  get(key: string, useLocalStorage = false): Promise<string | null> {
-    if (!this.values_[key]) {
-      this.values_[key] = new Promise((resolve) => {
+  async get(baseKey: string, useLocalStorage = false): Promise<string | null> {
+    // The old version of storage key without publication identifier.
+    // To be deprecaed in favor of the new version of key.
+    const oldKey = this.getStorageKeyWithoutPublicationId_(baseKey);
+    // The new version of storage key with publication identifier.
+    const newKey = this.getStorageKeyMaybeWithPublicationId_(baseKey);
+
+    const valueWithNewKey = await this.getInternal_(newKey, useLocalStorage);
+    if (valueWithNewKey !== null) {
+      return valueWithNewKey;
+    } else {
+      return this.getInternal_(oldKey, useLocalStorage);
+    }
+  }
+
+  private getInternal_(
```

Comment:
Is there a reason this needs to be a promise? Seems like you could just `return storage.getItem(finalKey)` or null.

Reply:
ditto on the set and delete methods

Reply:
It was copied over to encapsulate the original implementation to deal with storage interaction and caching. It looks like it caches the promises for retrieveing values in the 'values_' variable, not directly the values themselves.
I don't have context on why it's originally implemented to cache promises, but I think maintaining consistent return types for these internal methods (returning a promise) aligns with the cached values and simplifies the logic as it can just return the cached value if present.
I'm open to suggestions for refactoring this code. If you have any ideas for improvement, I'd be happy to follow up.